python/ray/data/read_api.py

```diff
@@ -1659,13 +1666,36 @@ def read_tfrecords(
             By default, the number of output blocks is dynamically decided based on
             input data size and available resources. You shouldn't manually set this
             value in most cases.
-
+        tfx_read_batch_size: An int representing the number of consecutive elements of
+            this dataset to combine in a single batch when tfx-bsl is used to read
+            the tfrecord files.
+        tfx_read_auto_infer_schema: Toggles the schema inference applied; applicable
+            only if tfx-bsl is used and tf_schema argument is missing.
+            Defaults to True.
     Returns:
         A :class:`~ray.data.Dataset` that contains the example features.
 
     Raises:
         ValueError: If a file contains a message that isn't a ``tf.train.Example``.
     """
+    import platform
+
+    tfx_read = False
```

Comment:
Hi @martinbomio, final requesst - can we add a config in `DataContext` - https://github.com/ray-project/ray/blob/master/python/ray/data/context.py . So user can disable the tfx read code path, in case they run into any bugs? By default, the config can set to true.

```py
tfx_read = DataContext.get_current().enable_tfrecord_tfx_read
```

Reply:
@c21 done in https://github.com/ray-project/ray/pull/42277/commits/413c1f01f5ca616b79e31ccc8a050bb88b6d1a5d